src/status_im/communities/core.cljs

```diff
@@ -555,19 +555,28 @@
   [community-id]
   (re-frame/dispatch [::fetch-requests-to-join community-id]))
 
+(defn update-activity-center-notification
+  [id]
+  [{:dispatch [:activity-center.notifications/mark-as-read id]
+    :ms       100}])
+
 (rf/defn request-to-join-accepted
   {:events [::request-to-join-accepted]}
   [{:keys [db] :as cofx} community-id request-id response-js]
-  (rf/merge cofx
-            {:db (update-in db [:communities/requests-to-join community-id] dissoc request-id)}
-            (handle-response response-js)))
+  (rf/merge
+   cofx
+   {:db             (update-in db [:communities/requests-to-join community-id] dissoc request-id)
+    :dispatch-later (update-activity-center-notification request-id)}
+   (handle-response response-js)))
 
 (rf/defn request-to-join-declined
   {:events [::request-to-join-declined]}
   [{:keys [db] :as cofx} community-id request-id response-js]
-  (rf/merge cofx
-            {:db (update-in db [:communities/requests-to-join community-id] dissoc request-id)}
-            (handle-response response-js)))
+  (rf/merge
+   cofx
+   {:db             (update-in db [:communities/requests-to-join community-id] dissoc request-id)
+    :dispatch-later (update-activity-center-notification request-id)}
```

Comment:
are you sure later is still required ? it could be just `:dispatch [:activity-center.notifications/mark-as-read id]`

Reply:
Yes, @flexsurfer - I tried those as well initially when I work on moving the event to `on-success` arguments. No luck in fixing the bug. A slight delay in dispatching the `:activity-center.notifications/mark-as-read` works.

Reply:
thats weird

Reply:
@ilmotta any thoughts?

Reply:
@smohamedjavid could you pls rollback then, because no need to hide dispatch in the function

Reply:
I'm honestly not sure, but the dispatch later shouldn't be necessary as @flexsurfer said.

Well, I have a guess @smohamedjavid, but it's hard without trying some things out, and it's hard to explain. Let me try...

The event to mark a notification as *read* receives a notification ID, and then get it from app DB (https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/activity_center/events.cljs#L131). The fact that it gets from the app DB means that it could be getting an outdated reference to the notification map while the other event hasn't completed. When you use `dispatch-later` you're giving re-frame a chance to update the app DB first.

So it's likely something to do with the fact that these events are seeing a different state stored in the app DB, and that's why one seem to replace the other's work and thus cause the UI bug.

Reply:
I think it would be valuable to understand why this problem happens, otherwise we're missing the chance of understanding something quite fundamental in terms of state management. I'm super curious too.